### PR TITLE
Upgrade `yargs` to fix vulnerability issue

### PR DESCRIPTION
**Pull request by jjloneman**
- https://snyk.io/test/npm/webpack-dev-server/3.11.0
- https://github.com/webpack/webpack-dev-server/issues/2854

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes. Added test in `cli.test.js` to demonstrate that updated `yargs` module properly displays usage statement.

Additionally, the existing tests which pass command-line options to the `webpack-dev-server` cli in `cli.test.js` pass which prove that the updated `yargs` module functions as expected.

### Motivation / Use-Case

Fix prototype pollution vulnerability issue in `yargs`.

See https://github.com/webpack/webpack-dev-server/issues/2854.

### Breaking Changes

No breaking changes.

### Additional Info

N/A

### CODE REVIEW

**Comment by evilebottnawi** [comment]
Breaking change and we can't update it

File: package.json [hunk 1]
```diff
@@ -2,7 +2,9 @@
   "name": "webpack-dev-server",
   "version": "3.11.0",
   "description": "Serves a webpack app. Updates the browser on changes.",
-  "bin": "bin/webpack-dev-server.js",
+  "bin": {
+    "webpack-dev-server": "bin/webpack-dev-server.js"
```
Comment (jjloneman) [inline package.json:6]: Just realized that `npm install` automatically changed this line for me. Running `webpack-dev-server` as a binary still functions the same (no breaking changes - https://docs.npmjs.com/cli/v6/configuring-npm/package-json#bin).

I can revert this line if desired.

**Comment by jjloneman** [comment]
> Breaking change and we can't update it

Is there a reason why it can't be updated?

**Comment by evilebottnawi** [comment]
Because https://github.com/webpack/webpack-dev-server/blob/00ca2a4d1abcc52477c0fbea91db1eb3141e0d6c/package.json#L16

**Comment by jjloneman** [comment]
@evilebottnawi Ah, makes sense. I'll open an issue in `yargs` to see if they can make a patch fix for `y18n` v4 for node 
6 then, then I'll update my PR with the fixed compatible version if they update it.

**Comment by alexander-akait** [comment]
Thanks for the PR, fixed in master, today we will release `webpack-dev-server@4-beta.0` (I will do stable release on the next week), anyway it is always should be very stable (we just move some options in other places)

**Pull request closed without merging**